### PR TITLE
feat: add feerate type

### PR DESCRIPTION
**Pull request by reez**
Introduces a new [FeeRate](https://docs.rs/bitcoin/0.32.2/bitcoin/struct.FeeRate.html) type.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1,13 +1,47 @@
...
+use error::FeeRateError;
 use error::ParseAmountError;
 
 #[macro_use]
 mod macros;
 pub mod error;
 pub use bitcoin::Network;
 
+#[derive(Clone, Debug)]
+pub struct FeeRate(pub(crate) BitcoinFeeRate);
```
Comment (rustaceanrob) [inline src/lib.rs:14]: I don't know if I understand the advantage in making this `crate` visible.

Comment (reez) [reply src/lib.rs]: totally, will remove `(crate)`

**Comment by thunderbiscuit** [comment]
@reez would you mind rebasing this?

**Comment by reez** [comment]
> @reez would you mind rebasing this?

👍 this is rebased now

**Pull request merged**